server/channels/app/plugin_requests.go

```diff
@@ -146,15 +147,31 @@ func (ch *Channels) servePluginRequest(w http.ResponseWriter, r *http.Request, h
 		token = r.URL.Query().Get("access_token")
 	}
 
+	// If MFA is required and user has not activated it, we wipe the token.
+	app := New(ServerConnector(ch))
+	rctx := request.EmptyContext(ch.srv.Log()).WithPath(r.URL.Path)
+
+	// The appErr is later used at L175.
```

Comment:
nit: Are we inferring something from the `appErr` that we could explicitly assign here instead? Having to do an `mfaAppErr` in between suggests there's something to improve.

Reply:
The `mfaAppErr` is an unfortunate necessity to avoid shadowing :(

Reply:
Can't we solve that trivially though?
```
appErr := app.MFARequired(rctx); 
if appErr != nil {
```

Reply:
Sure, but it's a tradeoff. It lengthens the lifetime of `appErr` variable further. So slightly higher chances of nil dereferencing. Although in this case, I think it's a worthy tradeoff. Let's do it.